tests/unit/widget/widget_classes.js

```diff
@@ -0,0 +1,137 @@
+(function( $ ) {
+
+module( "widget factory", {
+	teardown: function() {
+		if ( $.ui ) {
+			delete $.ui.testWidget;
+			delete $.fn.testWidget;
+		}
+	}
+});
+
+TestHelpers.testJshint( "widget" );
+
+$.widget( "ui.classesWidget", {
```

Comment:
The leftover part is fixed by fixing the teardown above ( which I did ). the only parts that are not needed in both tests that use it are the `toggleClasses` and `removeClasses` methods it seemed overkill to copy and paste most of this since where it goes onto `$.ui` its going to have to be manually cleaned up in the tear down anyway.